pkg/apis/m3dboperator/v1/cluster.go

```diff
@@ -202,6 +202,10 @@ type ClusterSpec struct {
 	// +optional
 	ContainerResources corev1.ResourceRequirements `json:"containerResources" yaml:"containerResources"`
 
+	// PersistentVolumeClaim claims a PersistentVolume for cluster storage;
+	// volumes are dynamically provisioned by when StorageClass is defined & applied
+	PersistentVolumeClaim corev1.PersistentVolumeClaim `json:"volumeClaimTemplate" yaml:"volumeClaimTemplate"`
```

Comment:
Can we make this an array of claims? I could foresee a setup where we might want to have something like the commit logs on different storage than the data files since they have distinct access patterns. Also I noticed the struct field is `PersistentVolumeClaim` but the json/yaml is `volumeClaimTemplate`.

Reply:
Yeah I was thinking about the array of claims as well, I'll implement that.

So within the KubeAPI the stuff specified in `volumeClaimTemplates` "turns into" a `PersistentVolumeClaim` corev1 API object -- in my opinion I think it makes the most sense to still name it `PersistentVolumeClaim` because that's what actually dynamically provisions the storage whereas the templates are moreso tied to the definition of a manifest, and that's what the methods are named after (such as `corev1.PersistentVolumeCLaimLister` and what not). I can add a clarifying comment if that helps